src/main/java/wansbot/WansBot.java

```diff
@@ -0,0 +1,287 @@
+
+    /**
+     * Throws InputEmptyException when user doesn't input anything after command word.
+     */
+    protected static void emptyInput(String userInput) throws InputEmptyException {
+        if (userInput.strip().equalsIgnoreCase("todos")
+                || userInput.strip().equalsIgnoreCase("deadline")
+                || userInput.strip().equalsIgnoreCase("event")
+                || userInput.strip().equalsIgnoreCase("mark")
+                || userInput.strip().equalsIgnoreCase("unmark")
+                || userInput.strip().equalsIgnoreCase("remove")) {
+            throw new InputEmptyException(userInput);
+        }
+    }
+
+    /**
+     * Throws NumberFormatException and NotANumMarkingException when a non-number is inputted and when it's not
+     * a valid task number in the userTaskList.
+     */
+    private static void notNumInput(String userInput, int taskListSize) throws NumberFormatException,
+            NotANumMarkingException {
+        if (userInput.startsWith("unmark")) {
+            int posTask = Integer.parseInt(userInput.substring(7));
+            if (posTask > taskListSize || posTask < 1) {
+                throw new NotANumMarkingException(posTask);
+            }
+        } else if (userInput.startsWith("mark")) {
+            int posTask = Integer.parseInt(userInput.substring(5));
+            if (posTask > taskListSize || posTask < 1) {
+                throw new NotANumMarkingException(posTask);
+            }
+        } else if (userInput.startsWith("remove")) {
+            int posTask = Integer.parseInt(userInput.substring(7));
+            if (posTask > taskListSize || posTask < 1) {
+                throw new NotANumMarkingException(posTask);
+            }
+        }
+    }
+
+    /**
+     * Throws InputEmptyException when user does not input a deadline after "deadline /by".
+     */
+    private static void missingInputDeadline(String userInput) {
+        String[] splitUser = userInput.split(" /by ", 2);
+        if (splitUser.length < 2) {
+            throw new InputEmptyException(userInput, "/by");
+        }
+    }
+
+    /**
+     * Throws custom InputEmptyException for events when either missing from date after /from, missing to date after
+     * /to or missing both.
+     */
+    protected static void missingInputEvent(String userInput) {
+        String[] splitUserStartDate = userInput.split(" /from ", 3);
+        if (splitUserStartDate.length < 2) {
+            throw new InputEmptyException(userInput, "/from");
+        }
+        String[] splitUserEndDate = splitUserStartDate[1].split(" /to ", 2);
+        if (splitUserEndDate.length < 2) {
+            throw new InputEmptyException(userInput, "/to");
+        }
+    }
+
+    /**
+     * Handles marking tasks function by specifying how caught exceptions are dealt with. Prints to console what
+     * user needs to correct for the command to work.
+     */
+    private static void markTasks(String userInput) {
```

Comment:
Making "Tasks" plural might confuse some users thinking they can pass in a list of Task indexes

Reply:
Oh yes, I should've thought of that. Thanks for pointing that out!